src/components/Task.jsx

```diff
@@ -0,0 +1,41 @@
+import PropTypes from 'prop-types';
+
+const Task = ({ task, setnewTask }) => {
+  const toggleTodo = (id, completed) => {
+    setnewTask(currentTasks => {
+      return currentTasks.map(task => {
+        if (task.id === id) {
+          return { ...task, completed }
+        }
+
+        return task;
+      })
+    })
+  }
```

Comment:
this function doesn't work 
you do something like this 
`f (task.id === id) {
          return { ...task, completed:!completed }
        }`